interchaintest/fee_middleware_test.go

```diff
@@ -0,0 +1,199 @@
+
+	// register CounterpartyPayee
+	cmd := []string{
+		"tx", "register-counterparty",
+		chainA.Config().Name,
+		channelA.ChannelID,
+		"transfer",
+		rlyAddressA,
+		rlyAddressB,
+	}
+	_ = r.Exec(ctx, eRep, cmd, nil)
+	require.NoError(t, err)
+
+	// Query the relayer CounterPartyPayee on a given channel
+	query := []string{
+		chainA.Config().Bin, "query", "ibc-fee", "counterparty-payee", channelA.ChannelID, rlyAddressA,
+		"--chain-id", chainID_A,
+		"--node", chainA.GetRPCAddress(),
+		"--home", chainA.HomeDir(),
+		"--trace",
+	}
+	_, _, err = chainA.Exec(ctx, query, nil)
+	require.NoError(t, err)
+
+	// Get initial account balances
+	userAOrigBal, err := chainA.GetBalance(ctx, userAddressA, chainA.Config().Denom)
+	require.NoError(t, err)
+	require.Equal(t, userFunds, userAOrigBal)
+
+	userBOrigBal, err := chainB.GetBalance(ctx, userAddressB, chainB.Config().Denom)
+	require.NoError(t, err)
+	require.Equal(t, userFunds, userBOrigBal)
+
+	rlyAOrigBal, err := chainA.GetBalance(ctx, rlyAddressA, chainB.Config().Denom)
+	require.NoError(t, err)
+	require.Equal(t, userFunds, rlyAOrigBal)
+
+	rlyBOrigBal, err := chainB.GetBalance(ctx, rlyAddressB, chainB.Config().Denom)
+	require.NoError(t, err)
+	require.Equal(t, userFunds, rlyBOrigBal)
+
+	// send tx
+	const txAmount = 1000
+	transfer := ibc.WalletAmount{Address: userAddressB, Denom: chainA.Config().Denom, Amount: txAmount}
+	_, err = chainA.SendIBCTransfer(ctx, channelA.ChannelID, userAddressA, transfer, ibc.TransferOptions{})
+	require.NoError(t, err)
+
+	// Incentivizing async packet by returning MsgPayPacketFeeAsync
+	packetFeeAsync := []string{
+		chainA.Config().Bin, "tx", "ibc-fee", "pay-packet-fee", "transfer", channelA.ChannelID, "1",
+		"--recv-fee", fmt.Sprintf("1000%s", chainA.Config().Denom),
+		"--ack-fee", fmt.Sprintf("1000%s", chainA.Config().Denom),
+		"--timeout-fee", fmt.Sprintf("1000%s", chainA.Config().Denom),
+		"--chain-id", chainID_A,
+		"--node", chainA.GetRPCAddress(),
+		"--from", walletA.FormattedAddress(),
+		"--keyring-backend", "test",
+		"--gas", "400000",
+		"--yes",
+		"--home", chainA.HomeDir(),
+	}
+	_, _, err = chainA.Exec(ctx, packetFeeAsync, nil)
+	require.NoError(t, err)
+
+	// start the relayer
+	err = r.StartRelayer(ctx, eRep, pathChainAChainB)
+	require.NoError(t, err)
+
+	t.Cleanup(
+		func() {
+			err := r.StopRelayer(ctx, eRep)
+			if err != nil {
+				t.Logf("an error occured while stopping the relayer: %s", err)
+			}
+		},
+	)
+
+	// Wait for relayer to run
+	err = testutil.WaitForBlocks(ctx, 5, chainA, chainB)
+	require.NoError(t, err)
+
+	// Assigning denom
+	chainATokenDenom := transfertypes.GetPrefixedDenom(channelA.PortID, channelA.ChannelID, chainA.Config().Denom)
+	chainADenomTrace := transfertypes.ParseDenomTrace(chainATokenDenom)
+
+	// Get balances after the fees
+	chainABal, err := chainA.GetBalance(ctx, userAddressA, chainA.Config().Denom)
+	require.NoError(t, err)
+	require.Equal(t, userAOrigBal-(txAmount+1000), chainABal)
```

Comment:
Hmm I would expect this assertion to be `chainABal = userAOrigBal - txAmount - feeAmount` which I would figure the amount in fees that is deducted should be `1000` for the Recv

Since `txAmount` is `1000` this is essentially saying that `userA`'s new balance should be equal to whatever it was before the tx was sent